packages/playwright-test/src/index.ts

```diff
@@ -218,13 +220,12 @@ export const test = _baseTest.extend<TestFixtures, WorkerFixtures>({
       options.permissions = permissions;
     if (proxy !== undefined)
       options.proxy = proxy;
-    if (storageState !== undefined) {
+    if (storageStateName !== undefined) {
+      const value = await test.info().storage().get(storageStateName);
+      if (value)
```

Comment:
Let's throw otherwise? Now, with an explicit option, we can be strict to avoid surprises.